### DISCUSSION TITLE
External Commissioning into Thread Domain with Thread 1.2 Commercial extensions

### DISCUSSION THREAD

**Discussion by arnoldlv1976** [openthread/openthread#9082] [Q&A] [answered]
Hello OpenThread community!

Actually my general question is about Thread 1.2 Commercial Extensions: How can I securely registry new devices (we have hundreds Thread devices) into a Thread Domain with zero-touch? What is the software architecture of the decision and what applications should I set up? What should be done from Thread device part? Is it opensource decision or should be bought some license to use it? 

To specify it I would ask more questions:
1.  Should I set up OpenThread Registrar (https://github.com/openthread/ot-registrar) server for it? How can I set it up understandable way? Is it opensource approach or need to buy some license? Is it still supportable?  How to configure Thread devices that it can find and connect to Registrar (TRI)? What is Thread Registrar Interface where is it described (I faced with a bad ref to TRI project)?

2. How  to do commissioning a Thread device using external app ot-commissioner (https://github.com/openthread/ot-commissioner) and using it in Commercial Commissioning Mode (CCM configuration)? What should be done on the Thread device side? Is There any detail information or example how to set up the Commissioner in CCM mode?

3. What can be architecture of a minimal enterprise decision, I need only that zero-touch commissioning since I have hundreds of Thread devices ( in Thread domain)?

Regards
Dima

**Comment by jwhui** [comment, marked as answer]
Unfortunately, the specific commercial commissioning features you are referring to are not yet finalized in the Thread specification. As such, the implementation is also not complete, which is why you are not able to find all the answers you are looking for. If you are interesting in contributing to the development of commercial commissioning features, I encourage you to join the Thread Group.
